inst/include/cpp11/doubles.hpp

```diff
@@ -134,33 +134,44 @@ typedef r_vector<double> doubles;
 
 }  // namespace writable
 
+template <>
+inline double na() {
+  return NA_REAL;
+}
+
+template <>
+inline bool is_na(const double& x) {
+  return ISNA(x);
+}
+
+// forward declarations
 typedef r_vector<int> integers;
 
+template <>
+int na();
```

Comment:
This forward declaration is required because of `is_na(el)` in `as_doubles()` below. There is no integer specialization for `is_na()`, but the default implementation of that template function does `x == na<T>()`, so we need the integer specialization of `na<T>()` here.